src/main/scala/com/github/unchama/generic/effect/MonadThrowExtra.scala

```diff
@@ -0,0 +1,32 @@
+package com.github.unchama.generic.effect
+
+import cats.MonadError
+import cats.effect.MonadThrow
+
+object MonadThrowExtra {
+
+  import cats.implicits._
+
+  def retryUntilSucceeds[F[_]: MonadThrow, A](fa: F[A])(limit: Int): F[A] = {
```

Comment:
`limit >= 1` を要求した方が良さそう

```suggestion
  def retryUntilSucceeds[F[_]: MonadThrow, A](fa: F[A])(limit: Int): F[A] = {
    require(limit >= 1)
```